src/components/atoms/ButtonBuy.tsx

```diff
@@ -138,7 +140,8 @@ export default function ButtonBuy({
   type,
   priceType,
   algorithmPriceType,
-  algorithmConsumableStatus
+  algorithmConsumableStatus,
+  isWhitelisted
```

Comment:
Are you sure `algortihmConsumableStatus` is usable in here? Even if my wallet is not whitelisted and I can't start a compute job, the `algortihmConsumableStatus` value is always **0** for me, and if I log the result of the `ocean.assets.isConsumable()` function (the one used to generate `algortihmConsumableStatus`) I get an object like this `{status: 0, message: 'All good', result: true}`.
Instead if I use `ocean.assets.checkCredential()` I get the response I'm looking for: `{status: 2, message: 'Access is denied, your wallet address is not found on allow list', result: false}`

Reply:
Alright, thanks for clarifying. I believe we should rather fix `algorithmConsumableStatus` then, since this should already pass the correct status number. Maybe have a look at the reference again ([Ocean Market v4](https://github.com/oceanprotocol/market/blob/64255f2b371ec53713d444819c4732966f7193f5/src/components/%40shared/ButtonBuy/index.tsx#L97)) to see if it is working correctly there.